algorithms/kernel/adaboost/adaboost_model_fpt.cpp

```diff
@@ -29,23 +29,28 @@ namespace algorithms
 {
 namespace adaboost
 {
-
+namespace interface2
+{
 /**
  * Constructs the AdaBoost model
  * \tparam modelFPType  Data type to store AdaBoost model data, double or float
  * \param[in] dummy     Dummy variable for the templated constructor
  * \DAAL_DEPRECATED_USE{ Model::create }
  */
 template <typename modelFPType>
-DAAL_EXPORT Model::Model(size_t nFeatures, modelFPType dummy) : boosting::Model(nFeatures)
+DAAL_EXPORT Model::Model(size_t nFeatures, modelFPType dummy) :
+    _nFeatures(nFeatures),
+    _models(new data_management::DataCollection())
 {
     _alpha = data_management::NumericTablePtr(new data_management::HomogenNumericTable<modelFPType>(NULL, 1, 0));
 }
 
 template <typename modelFPType>
-DAAL_EXPORT Model::Model(size_t nFeatures, modelFPType dummy, services::Status &st) : boosting::Model(nFeatures, st)
+DAAL_EXPORT Model::Model(size_t nFeatures, modelFPType dummy, services::Status &st) :
+    _nFeatures(nFeatures),
+    _models(new data_management::DataCollection())
 {
-    if (!st) { return; }
+    if (!_models) { st.add(services::ErrorMemoryAllocationFailed); }
     _alpha = data_management::HomogenNumericTable<modelFPType>::create(NULL, 1, 0, &st);
```

Comment:
Could it be moved into initialization?

Reply:
no, there is no initialization function for adaboost model